services/src/main/java/org/keycloak/broker/oidc/OIDCIdentityProvider.java

```diff
@@ -196,6 +201,17 @@ public Response keycloakInitiatedBrowserLogout(KeycloakSession session, UserSess
         }
     }
 
+    protected String extractLoginHintFromIdToken(String idToken) {
```

Comment:
Instead of parsing the ID Token, would it be enough to get the username from `org.keycloak.events.Details#IDENTITY_PROVIDER_USERNAME`?

Reply:
Not in case of MS EntraID...
In general, it's not clearly specified, what exactly should be put into the optional `logout_hint` parameter.

For Microsoft, they need to have the `login_hint` value from the id token, as mentioned [here](https://learn.microsoft.com/en-us/entra/identity-platform/v2-protocols-oidc#send-a-sign-out-request) (this link is also mentioned in the corresponding issue #33008):
> ...and use the value of the login_hint optional claim as the logout_hint parameter. Don't use UPNs or phone numbers as the value of the logout_hint parameter.

I'm currently not aware of other IdPs actually using this parameter.
Parsing it from the id token is the most generic way I can currently think of.

Reply:
Sounds like we should introduce this behavior only for `org.keycloak.social.microsoft.MicrosoftIdentityProvider` then?

Having this in `OIDCIdentityProvider` suggests it applies to other vendors, and I think that is not the case. AFAIK, the ID Token in the specification does not define a `login_hint` claim, does it?

Reply:
Some dedicated support of EntraID makes definitely sense, as this is one of the most federated IdPs for companies and corporates.

I don't know, if a dedicated provider makes sense in terms of maintainability. In general, EntraID conforms mostly to the OIDC spec. The `logout_hint` behavior is the first, which is - AFAIK - interpreted opinionatedly.
Would it make sense to extend the mapper functionality also towards logout? E.g. one could create a mapper to map something from user, session, tokens, etc. to logout params!? Or is this too complicated to implement?

If the mapper approach isn't feasible, I'd also vote for some EntraID provider like you mentioned with your no. 2